typescript/sdk/src/providers/transactions/transformer/ethersV5/EV5InterchainAccountTxTransformer.ts

```diff
@@ -25,29 +27,36 @@ export class EV5InterchainAccountTxTransformer
 
   public async transform(
     ...txs: PopulatedTransaction[]
-  ): Promise<PopulatedTransaction[]> {
+  ): Promise<ethers.PopulatedTransaction[]> {
     const txChainsToInnerCalls: Record<ChainName, CallData[]> = {};
 
     txs.map(({ to, data, value, chainId }: PopulatedTransaction) => {
-      assert(to, 'Invalid PopulatedTransaction: Missing to field');
-      assert(data, 'Invalid PopulatedTransaction: Missing data field');
-      assert(chainId, 'Invalid PopulatedTransaction: Missing chainId field');
       const txChain = this.multiProvider.getChainName(chainId);
       if (!txChainsToInnerCalls[txChain]) txChainsToInnerCalls[txChain] = [];
       txChainsToInnerCalls[txChain].push({ to, data, value });
     });
 
-    const transformedTxs: Promise<PopulatedTransaction>[] = [];
+    assert(
+      this.props.config.localRouter,
+      'Invalid AccountConfig: Cannot retrieve InterchainAccount.',
+    );
```

Comment:
assert this in constructor instead of here?

Reply:
definitely– moved up